### DISCUSSION TITLE
Task 4 Lab 2.04. Duda con la ecuacion (SOLUCIONADO!)

### DISCUSSION THREAD

**Discussion by rbecerrav** [rramosp/ai4eng.v1#25] [02 - PYTHON]
Profesor Raul buenos dias. tengo una duda en el ultimo punto donde hay que aplicar la ecuacion matematica para saber la posicion de la carta. a continuacion muestro el ejercicio. y al final se puede apreciar a manera de ejemplo la prueba que hago de la ecuacion. la cual no me esta arrojando el valor esperado
muchas gracias

## Task 4: Run the trick using the math

Given:

- $n_h$: A number of heaps
- $c_h$: the number of cards per heap
- $i$: the position of a card 

The new position of the card after one cycle of making and collecting the heaps will be:

$$c_h(n_h\div2)+i\div n_h$$

Complete the following function so that it has the same functionality of the previous task, but applying only this formula without using the simulation above. You **MUST NOT ADD** or remove lines from the function skeleton below, **ONLY** fill in the `...`

You must get the same results as the previous task.

**HINT**: Use `np.argwhere` to get the initial position of the card in the deck

#Profe Raul. hago la prueba con un mazo de 12 y con una posicion inicial de la 
#carta seleccionada en 1. y luego de la primer recogida la posicion en que 
#queda la carta es la 8 pero la ecuacion me arroja 4 
`4*(3//2)+(1//3)  = 4`

**Comment by maomiranda30** [comment]
Buenas noches disculpe estoy barado en ese mismo ejercicio, me puedes orientar como llegaste a la solución

**Reply by rbecerrav** [reply]
Hola compañero yo estaba un poco confundido porque estaba ingresando mal la ecuacion, al final la funcion me quedo de la siguiente manera 
`def mrun(c, picked_card, n_heaps=3):
    assert len(c)%n_heaps==0, "the number of heaps must be a divisor of the deck length"
    
    ch = len(c)//n_heaps # cards per heap
    nh = n_heaps
    
    i = int(np.argwhere(c==picked_card)) # initial position of the card on the deck c
    p1 = (ch*(nh//2)+((i//nh)))  # the position of the card after the first round
    p2 = (ch*(nh//2)+((p1//nh)))  # the position of the card after the second round
    p3 = (ch*(nh//2)+((p2//nh)))  # the position of the card after the last round
    
    return p3`

**Comment by rramosp** [comment]
hola a todos, gracias rbecerrav por tu contribución .... queremos precisamente fomentar que entre todos nos ayudemos tanto para mejorar el curso como crear comunidad entre todos!!!

**Reply by rbecerrav** [reply]
con mucho gusto
